aea/crypto/ethereum.py

```diff
@@ -268,7 +262,8 @@ def send_raw_transaction(self, tx_signed, tx_type) -> Optional[str]:
                 result = self._api.eth.getTransactionReceipt(hex_value)
                 logger.info("transaction validated - exiting")
                 if result.contractAddress is not None:
-                    self._contract_address = result.contractAddress
+                    # Checking the address so I can pass it manually for now.
+                    logger.info(result.contractAddress)
```

Comment:
why don't you just return the transaction digest `tx_digest` and the `result`? It seems the generic use case is that I want to get these two values back and then if it happens to be a transaction associated with contract deployment the contract address can be recovered in the business logic where this is called.

Reply:
So can we consider to always returns the result and if we need the `tx_hash` just get it from the AttributeDictionary? The Dictionary is in the form of 
```
info: AttributeDict({
   'blockHash': HexBytes('...'),
   'blockNumber': 7418505, 
   'contractAddress': '0x03D2B44B573418B8ACfe53810379FfdF6E21E0F1',                 'cumulativeGasUsed': 5185623, 
   'from': '0x21795d753752ccc1ac728002d23ba33cbf13b8b0', 
   'gasUsed': 3603991, 
   'logs': [],
   'logsBloom': HexBytes('a lot of 0'), 
   'status': 1, 'to': None, 
   'transactionHash':         HexBytes('0xa1920cb04cbd645f7f81bd3b745cd847e442a33335a7b3dc18ebfdfb36602af6'),
   'transactionIndex': 28})
```

Reply:
yes ok